packages/flutter/lib/src/painting/image_resolution.dart

```diff
@@ -240,19 +240,39 @@ class AssetImage extends AssetBundleImageProvider {
     // TODO(ianh): implement support for config.locale, config.textDirection,
     // config.size, config.platform (then document this over in the Image.asset
     // docs)
-    return _findNearest(mapping, config.devicePixelRatio!);
+    return _findNearest(mapping, config.devicePixelRatio!, config.platform);
   }
 
   // Return the value for the key in a [SplayTreeMap] nearest the provided key.
-  String? _findNearest(SplayTreeMap<double, String> candidates, double value) {
+  String? _findNearest(SplayTreeMap<double, String> candidates, double value, TargetPlatform? platform){
     if (candidates.containsKey(value))
       return candidates[value]!;
     final double? lower = candidates.lastKeyBefore(value);
     final double? upper = candidates.firstKeyAfter(value);
+
+    // Desktop devices frequently have high resolution screens but low DPR
+    // values. Until https://github.com/flutter/flutter/issues/63122 is resolved,
+    // improve the situation slightly by preferring the higher DPR image.
+    bool roundUp;
+    switch (platform) {
```

Comment:
I don't think this is the right idiom.

What we really care about is whether we're on desktop or not. Scratch that, what we really care about is whether we have a high physical size and a low DPR, which is why desktop becomes problematic for this.

The problem with using TargetPlatform this way is I suspect it will give us undesirable results in the following types of cases:

- A decently sized Chromebook with a web browser.
- A small, Linux based tablet or phone-sized screen.

Reply:
What do you think about doing this based on the following heuristic:

If the physical size is greater than 1080p && the DPR <= 1.5, round up?

Reply:
We could also look at just using the W3C criteria for media query large or not based on logical size.

Reply:
> A decently sized Chromebook with a web browser.

The originally filled issue was actually on a chromebook - they have the same screen problem.

> A small, Linux based tablet or phone-sized screen.

We do not as of yet run on these devices so it is difficult for me to say